### ISSUE TITLE
Addition of custom FIAT Currency value (Chilean Unidad de Fomento)

### ISSUE THREAD

**Issue by Goro2030** [btcpayserver/btcpayserver#7522]
I have a server in Chile, and I would like to use a local pseudo-currency that we have called "Chilean Unit of Account" (UF for short, Unidad de Fomento in spanish). We of course have the Chilean Peso ( CLP in BTCPay ), but some transactions are done in UF's. 

The UF has an official exchange rate into Chilean Pesos that's dictated by the government and it changes daily.

I can easily get the value of one UF by doing a search in google, or using the GoogleFinance API from within Google Sheets, but I was wondering if I can add a "custom FIAT currency" in BTCPayServer, and tell BTCPay how to find out it's quote?

I currently get it for other purposes with this API call:
=IMPORTJSON("https://mindicador.cl/api","/uf/valor","noHeaders")

**Comment by LodrikSatoshiar** [comment]
Hi @NicolasDorier, greetings from Satoshilemu.

I’d like to revive this old issue, as having UF support in BTCPay Server is becoming increasingly relevant for us in Chile.

Our BTCPay Server deployment has been expanding, with more users and merchants joining the platform, and we are increasingly receiving requests from users who need to display and calculate prices in UF.

The original issue mentioned a third-party API to retrieve the UF/CLP rate. However, the Central Bank of Chile provides an official and free API through its Statistical Database (BDE):

https://si3.bcentral.cl/Siete/es/Siete/API?respuesta=

The Unidad de Fomento (UF) is widely used in Chile as a unit of account for prices, contracts, real estate, financial services, and many other transactions. Its value in CLP is officially determined and published daily.

Adding the Central Bank of Chile as a rate source and supporting UF would allow BTCPay users in Chile to denominate prices and invoices in UF while receiving Bitcoin, using an official source for the UF/CLP conversion.

@NicolasDorier, would it be possible to consider adding this in an upcoming release? There is growing demand for this functionality among our users, and I’d be happy to help with testing from Chile if needed.

Thanks!

**Comment by NicolasDorier** [comment]
@LodrikSatoshiar yes, please open a PR on https://github.com/btcpayserver/btcpayserver you can probably vibe code that, once done, I will cleanup to make sure it's ok.

**Comment by Shuaibu78** [comment]
Hi @LodrikSatoshiar, are you already working on this? If not, I'd be happy to pick it up.

While looking into it I noticed two things:

UF already exists in BTCPay's currency list under its ISO 4217 code CLF.

BitPay's rate feed (already supported) publishes BTC/CLF (currently about 39,486 CLP per UF), so X_CLF = bitpay(X_CLF); should already work.

@NicolasDorier, which would you prefer?

Small PR: set bitpay as the recommended exchange for CLF so it works out of the box, plus a test.
Bigger PR: add a Banco Central de Chile provider for the official CLF/CLP rate. Their API needs registered credentials, and no current rate provider uses API keys, so I'd want your guidance on how to handle that.

@LodrikSatoshiar, could you check whether option 1 works for your merchants today? Setting the store currency to CLF with that rule should be enough.

**Comment by NicolasDorier** [comment]
@Shuaibu78
Indeed, we do not support API keys in providers, so do the small PR.

**Comment by Goro2030** [comment]
Great to see this moving @Shuaibu78 , thanks!

**Comment by Shuaibu78** [comment]
> [@Shuaibu78](https://github.com/Shuaibu78) Indeed, we do not support API keys in providers, so do the small PR.

okay thanks, working on this
